lib/takeaway.rb

```diff
@@ -0,0 +1,46 @@
+require_relative 'restaurant.rb'
+
+class Takeaway
+  
+  attr_reader :restaurant, :order_log
+  
+  def initialize(restaurant=Restaurant.new, order=Order.new)
+    @action ||= nil
+    @restaurant = restaurant
+    @order_log = order
+  end
+  
+  def menu
+    read_list
+    get_choice
+  end
+  
+  def get_choice
+    @action ? @action : @action = gets.chomp
+    display_menu
+  end
+  
+  def order(item, capacity=1)
+    return "Please choose an item in the menu" unless self.restaurant.menu.has_key?(item.capitalize)     #refactor for SRP 
```

Comment:
Metrics/LineLength: Line is too long. [123/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)<br>Style/DeprecatedHashMethods: `Hash#has_key?` is deprecated in favor of `Hash#key?`. (https://github.com/bbatsov/ruby-style-guide#hash-key)